zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -152,6 +152,45 @@ sealed trait Schema[A] {
 }
 
 object Schema extends TupleSchemas with RecordSchemas with EnumSchemas {
+
+  private def fuzzyReformat(str: String): String = str.replaceAll("[-_]", "").toLowerCase
+
+  @tailrec
+  private def isTypeMatch(schema: Schema[_], valueType: StandardType[_]): Boolean =
```

Comment:
Should be able to leverage the schema equality work recently added by @vigoo here.